src/nbl/video/IGPUDescriptorSet.cpp

```diff
@@ -140,7 +157,7 @@ void IGPUDescriptorSet::dropDescriptors(const IGPUDescriptorSet::SDropDescriptor
     const auto descriptorType = getBindingType(drop.binding);
 
 	auto* dstDescriptors = drop.dstSet->getDescriptors(descriptorType, drop.binding);
-	auto* dstSamplers = drop.dstSet->getMutableSamplers(drop.binding);
+	auto* dstSamplers = drop.dstSet->getMutableCombinedSamplers(drop.binding);
```

Comment:
you should skip this call and the contents of the if statement on line 166 if the `descriptorType` is not COMBINED

(there's a binary search or hashmap lookup involved, its potentially expensive)